pkg/api/unversioned/types.go

```diff
@@ -339,6 +339,8 @@ type APIResource struct {
 	Name string `json:"name"`
 	// namespaced indicates if a resource is namespaced or not.
 	Namespaced bool `json:"namespaced"`
+	// kind is the kind for the resource (e.g. 'Foo' is the kind for a resource 'foo')
+	Kind string `json:"kind"`
```

Comment:
As far as I know there is no affordance (currently) for serving APIResource objects for subresources.  If we wanted to do that we'd probably need to have a full sub-object here anyway, seems beyond the scope of this PR.